kod/util/system.kod

```diff
@@ -2720,6 +2726,28 @@ messages:
       return;
    }
 
+   RecreateShrineAllegiances()
+   "Resets shrine allegiances and power levels"
+   {
+      local i, n;
+
+      if plShrine_Allegiances <> $
+      {
+         n = 5;
+
+         for i in plShrines
+         {
+            Send(i,@SetAllegiance,#allegiance=(Nth(plShrine_Allegiances,n)));
```

Comment:
This is confusing that the list of allegiances is backward compared to the list of shrines.  Can you build up the list so that it's in the same direction?

Reply:
I've been racking my brain on this for a bit and between my fading knowledge and what I see in the manual, the only solution I could come up with is creating a temporary local list that I then use to map to `plShrine_Allegiances` in reverse order, so as to sync up with the order of `plShrines`. Is there a better way to go about this?

Reply:
You could iterate backward through plShrines in RecreateAllRooms(), rather than here, and then the lists would be in the same order.